autogen/agenteval/agent_eval.py

```diff
@@ -0,0 +1,101 @@
+import autogen
+from criterion import Criterion
+from critic_agent import CriticAgent
+from quantifier_agent import QuantifierAgent
+from subcritic_agent import SubCriticAgent
+from task import Task
+
+import os
+import sys
+from typing import Callable, Dict, Optional, Union
+
+
+def generate_criteria(
+    llm_config: Optional[Union[Dict, bool]] = None,
+    task: Task = None,
+    additional_instructions: str = "",
+    max_round=2,
+    use_subcritic: bool = False,
+):
+    """
+    Creates a list of criteria for evaluating the utility of a given task.
+    args:
+    - llm_config (dict or bool): llm inference configuration.
+    - task (Task): The task to evaluate.
+    - additional_instructions (str): Additional instructions for the criteria agent.
+    - max_round (int): The maximum number of rounds to run the conversation.
+    - use_subcritic (bool): Whether to use the subcritic agent to generate subcriteria.
+    returns:
+    - list: A list of Criterion objects for evaluating the utility of the given task.
+    """
+    critic = CriticAgent(
+        system_message=CriticAgent.DEFAULT_SYSTEM_MESSAGE + "\n" + additional_instructions,
+        llm_config=llm_config,
+    )
+
+    critic_user = autogen.UserProxyAgent(
+        name="critic_user",
+        max_consecutive_auto_reply=0,  # terminate without auto-reply
+        human_input_mode="NEVER",
+        code_execution_config={"use_docker": False},
+    )
+
+    agents = [critic_user, critic]
+
+    if use_subcritic:
+        subcritic = SubCriticAgent(
+            llm_config=llm_config,
+        )
+        agents.append(subcritic)
+
+    groupchat = autogen.GroupChat(
+        agents=agents, messages=[], max_round=max_round, speaker_selection_method="round_robin"
+    )
+    critic_manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)
+
+    critic_user.initiate_chat(critic_manager, message=task.sys_msg)
+    criteria = critic_user.last_message()
+    print(criteria["content"])
+    criteria = Criterion.parse_json_str(criteria["content"])
+    return criteria
+
+
+def quantify_criteria(
```

Comment:
this function could technically be part of quantifier_agent as a method - with some caveats. It might need to 'reset' the conversation every time to avoid quantifying with the previous calls.

Reply:
see critic_agent comment